events/schema.md

```diff
@@ -4,30 +4,157 @@ Events published to the bus have a standard schema that allows SDKs and services
+            "service": "article-store",
+            "name": "article-version",
+            "identifier": "1",
+        },
+        {
+            "service": "article-store",
+            "name": "article-run",
+            "identifier": "6551f09a-bd23-4f12-a039-f8fa78ef776a",
+        },
+        {
+            "service": "search",
+            "name": "indexing",
+            "identifier": "18b21fea-5a87-11e8-9c2d-fa7ae01bbebc",
+        },
+    ],
+}
+```
+
+### Example: downstream deposit service pushing an article to Crossref
+
+```
+{
+    "eventId": "448278c4-22d6-11e8-b467-0ed5f89f718b",
+    "happenedAt": "2018-03-08T12:00:00+00:00",
+    "aggregate": {
+        "service": "downstream-sample",
+        "name": "deposit",
+        "identifier": "2018-03-08/1", # daily attempts
+    },
+    "type": "downstream-crossref-started",
+    "data": {
+        ... 
+    },
+    "correlationIds": [
+        {
+            "service": "article-store",
+            "name": "article",
+            "identifier": "10627",
+        },
+        {
+            "service": "article-store",
+            "name": "article-version",
+            "identifier": "1",
+        },
+        {
+            "service": "article-store",
+            "name": "article-run",
+            "identifier": "6551f09a-bd23-4f12-a039-f8fa78ef776a",
+        },
+        {
+            "service": "downstream-sample",
+            "name": "deposit",
+            "identifier": "2018-03-08/1",
+        },
+    ],
+}
+```
+
+## Fields
+
+### `eventId`
+
+Unique for all events. It is a UUID version 1 generated with time and a node identifier.
+
+### `happenedAt` 
+
+Identifies when the event has happened (then it can get published, consumed, etc). It follows the `yyyy-mm-ddTHH:MM:SS+00:00` ISO 8601 format, and is expressed in UTC.
+
+### `aggregate`
+
+Identifies the unit of consistency that has produced the event: article run in the bot, article version in the articles store, podcast episode in journal-cms, an article's set of metrics in metrics, a single profile in profiles.
+
+All three information are required to uniquely identify it:
+
+- `service`: `[a-z\-]+`
+- `name`: `[a-z\-]+`
+- `identifier`: `.+` (possibly URL-friendly)
+
+### `type`
+
+Describes what the event is about so that they can be grouped or recognized. It follows the `[a-z\-\.]+` regular expression, with `.` separating hierarchical levels and `-` separating words in a single level (if there are multiple words).
+
+### `data` (optional)
+
+Is opaque here, and can contain anything as long as it follows naming conventions. Events with the same `type` should usually follow the same schema.
+
+### `correlationIds` (optional)
+
+Can be used to track events across services, grouping them by the kind of session that originated them or the data type they refer to. Could be an article run during ingestion, or a journal page creating a profile.
+
+Represents an array of dictionaries, each following the format of `aggregate`.
```

Comment:
Ordered?

Reply:
To be fair there is no clear order, loosely hierarchical or "order of appearance" but there may be cases in which it's ambiguous. Given this is JSON the alternative would be a dictionary of dictionaries, with something being used as key, does that work better?

Reply:
Think it's an unordered set, so array of objects is fine (just documentation that the order isn't important). Unless there's a natural key, which there isn't (being value objects).